CMakeLists.txt

```diff
@@ -84,6 +84,9 @@ if(SFML_BUILD_WINDOW)
     sfml_set_option(SFML_OPENGL_ES ${OPENGL_ES} BOOL "TRUE to use an OpenGL ES implementation, FALSE to use a desktop OpenGL implementation")
 endif()
 
+# add an option for building the test suite
+sfml_set_option(SFML_BUILD_TEST_SUITE TRUE BOOL "TRUE to build the SFML test suite, FALSE to ignore it")
```

Comment:
It's OFF by default in most libs, I think. So we'd better do the same ;)

Reply:
It will also indicate errors/problems when the test suite fails for someone. Unit tests are there to be run, not to be ignored. IMO it should stay enabled by default.

If we want to make unit tests part of our workflow, then we will have to explain to contributors how to enable tests, so that their PRs get accepted. Another annoying step to do that can be avoided.

The extra build time is only required for the initial build, especially when building the test suite "main file". Consequent builds are much faster.

Reply:
It's not just the build time, but also the run time, which could take any amount of time in the future once more (and more complex) tests are added. 

furthermore, it causes the build itself to fail meaning if a unit test fails for any reason the user won't be able to install SFML without fixing the bug or removing the test. 

I think it would be more reasonable to ask people to run tests before reporting an issue or sending a PR rather than blocking use of SFML until the issue is found and resolved